### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by svanherk**
With these changes:
- Node 18 will now be required (everyone currently using `@brightspace-ui/testing` is on node 18).
- For some reason, single quotes around the `--files` piece doesn't work (from a quick search, I don't think anyone is doing this except this repo)
- Dev mode was turned on by default in the runner and the dev-server
  - We already turn off `nodeResolve` in the dev server for the report
  - I've changed the default in the runner to be `{ exportConditions: ['default'] }`

### CODE REVIEW

**Comment by svanherk** [comment]
I'm going to test this in `core`, and we might want to wait until next week to merge this since I'm not confident it won't break something.

File: package.json
Comment (bearfriend) [inline package.json:12]: What was the issue with quotes for you? This can cause issues on POSIX system because it tries to pre-fill the wildcard.

Comment (svanherk) [reply package.json:12]: Error was:
```
Error: Could not find any test files with pattern(s): './test/browser/**/*.test.js',!**/node_modules/**/*
    at createTestSessions (C:\Desire2Learn\Instances\BrightspaceUI\testing\node_modules\@web\test-runner-core\dist\runner\createSessionGroups.js:55:19)
    at new TestRunner (C:\Desire2Learn\Instances\BrightspaceUI\testing\node_modules\@web\test-runner-core\dist\runner\TestRunner.js:32:115)
    at Object.startTestRunner [as start] (C:\Desire2Learn\Instances\BrightspaceUI\testing\node_modules\@web\test-runner\dist\startTestRunner.js:24:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runTests (file:///C:/Desire2Learn/Instances/BrightspaceUI/testing/bin/d2l-test-runner.js:47:2)
```
Maybe we need to trim the quotes off in the client?

Comment (bearfriend) [reply package.json:12]: Was that locally or in CI? I think I saw @dlockhart do this another way somewhere but I can't remember what it was. Escape the `*`, maybe?

Comment (svanherk) [reply package.json:12]: Locally. We ran into something similar here, but that was a totally different library. Seems odd.https://github.com/BrightspaceUI/core/pull/4222

Comment (dlockhart) [reply package.json:12]: Hmm, not recalling "the other way".

Comment (svanherk) [reply package.json]: Looks like escaped quotes work. Still not sure what exactly changed, but I assume this is what could be used to get around the POSIX system issue?

**Comment by svanherk** [comment]
This causes no vdiff changes in `core`: https://github.com/BrightspaceUI/core/pull/4247

**Review by dlockhart — APPROVED** [review]

**Pull request merged**
